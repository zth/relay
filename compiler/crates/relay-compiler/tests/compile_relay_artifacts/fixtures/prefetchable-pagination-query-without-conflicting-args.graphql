query prefetchablePaginationQueryWithoutConflictingArgsQuery($site: String) {
  node(id: "x") {
    ...prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment @arguments(site: $site)
    ...prefetchablePaginationQueryWithoutConflictingArgs_SecondFragment @arguments(site: $site)
  }
}

fragment prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment on Page
  @refetchable(queryName: "RefetchableFragmentQuery")
  @argumentDefinitions(
    count: {type: "Int", defaultValue: 10}
    cursor: {type: "ID"}
    site: {type: "String"}
  ) {
  name
  friends(after: $cursor, first: $count)
    @connection(key: "PaginationFragment_friends", prefetchable_pagination: true) {
    edges {
      node {
        id
        url(site: $site)
      }
    }
  }
}

fragment prefetchablePaginationQueryWithoutConflictingArgs_SecondFragment on User
  @argumentDefinitions(
    count: {type: "Int", defaultValue: 10}
    cursor: {type: "ID"}
    site: {type: "String"}
  ) {
    id
    friends(after: $cursor, first: $count)
        @connection(key: "PaginationFragment2_friends") {
        edges {
            node {
                id
                url(site: $site)
            }
        }
    }
}
