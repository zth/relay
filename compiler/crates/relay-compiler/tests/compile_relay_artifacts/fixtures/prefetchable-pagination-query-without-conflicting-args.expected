==================================== INPUT ====================================
query prefetchablePaginationQueryWithoutConflictingArgsQuery($site: String) {
  node(id: "x") {
    ...prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment @arguments(site: $site)
    ...prefetchablePaginationQueryWithoutConflictingArgs_SecondFragment @arguments(site: $site)
  }
}

fragment prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment on Page
  @refetchable(queryName: "RefetchableFragmentQuery")
  @argumentDefinitions(
    count: {type: "Int", defaultValue: 10}
    cursor: {type: "ID"}
    site: {type: "String"}
  ) {
  name
  friends(after: $cursor, first: $count)
    @connection(key: "PaginationFragment_friends", prefetchable_pagination: true) {
    edges {
      node {
        id
        url(site: $site)
      }
    }
  }
}

fragment prefetchablePaginationQueryWithoutConflictingArgs_SecondFragment on User
  @argumentDefinitions(
    count: {type: "Int", defaultValue: 10}
    cursor: {type: "ID"}
    site: {type: "String"}
  ) {
    id
    friends(after: $cursor, first: $count)
        @connection(key: "PaginationFragment2_friends") {
        edges {
            node {
                id
                url(site: $site)
            }
        }
    }
}
==================================== OUTPUT ===================================
{
  "fragment": {
    "argumentDefinitions": [
      {
        "defaultValue": 10,
        "kind": "LocalArgument",
        "name": "count"
      },
      {
        "defaultValue": null,
        "kind": "LocalArgument",
        "name": "cursor"
      },
      {
        "defaultValue": null,
        "kind": "LocalArgument",
        "name": "id"
      },
      {
        "defaultValue": null,
        "kind": "LocalArgument",
        "name": "site"
      }
    ],
    "kind": "Fragment",
    "metadata": null,
    "name": "RefetchableFragmentQuery",
    "selections": [
      {
        "alias": null,
        "args": [
          {
            "kind": "Variable",
            "name": "id",
            "variableName": "id"
          }
        ],
        "concreteType": null,
        "kind": "LinkedField",
        "name": "node",
        "plural": false,
        "selections": [
          {
            "args": [
              {
                "kind": "Variable",
                "name": "count",
                "variableName": "count"
              },
              {
                "kind": "Variable",
                "name": "cursor",
                "variableName": "cursor"
              },
              {
                "kind": "Variable",
                "name": "site",
                "variableName": "site"
              }
            ],
            "kind": "FragmentSpread",
            "name": "prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment"
          }
        ],
        "storageKey": null
      }
    ],
    "type": "Query",
    "abstractKey": null
  },
  "kind": "Request",
  "operation": {
    "argumentDefinitions": [
      {
        "defaultValue": 10,
        "kind": "LocalArgument",
        "name": "count"
      },
      {
        "defaultValue": null,
        "kind": "LocalArgument",
        "name": "cursor"
      },
      {
        "defaultValue": null,
        "kind": "LocalArgument",
        "name": "site"
      },
      {
        "defaultValue": null,
        "kind": "LocalArgument",
        "name": "id"
      }
    ],
    "kind": "Operation",
    "name": "RefetchableFragmentQuery",
    "selections": [
      {
        "alias": null,
        "args": [
          {
            "kind": "Variable",
            "name": "id",
            "variableName": "id"
          }
        ],
        "concreteType": null,
        "kind": "LinkedField",
        "name": "node",
        "plural": false,
        "selections": [
          {
            "alias": null,
            "args": null,
            "kind": "ScalarField",
            "name": "__typename",
            "storageKey": null
          },
          {
            "alias": null,
            "args": null,
            "kind": "ScalarField",
            "name": "id",
            "storageKey": null
          },
          {
            "kind": "InlineFragment",
            "selections": [
              {
                "alias": null,
                "args": null,
                "kind": "ScalarField",
                "name": "name",
                "storageKey": null
              },
              {
                "alias": null,
                "args": [
                  {
                    "kind": "Variable",
                    "name": "after",
                    "variableName": "cursor"
                  },
                  {
                    "kind": "Variable",
                    "name": "first",
                    "variableName": "count"
                  }
                ],
                "concreteType": "FriendsConnection",
                "kind": "LinkedField",
                "name": "friends",
                "plural": false,
                "selections": [
                  {
                    "alias": null,
                    "args": null,
                    "concreteType": "FriendsEdge",
                    "kind": "LinkedField",
                    "name": "edges",
                    "plural": true,
                    "selections": [
                      {
                        "alias": null,
                        "args": null,
                        "concreteType": "User",
                        "kind": "LinkedField",
                        "name": "node",
                        "plural": false,
                        "selections": [
                          {
                            "alias": null,
                            "args": null,
                            "kind": "ScalarField",
                            "name": "id",
                            "storageKey": null
                          },
                          {
                            "alias": null,
                            "args": [
                              {
                                "kind": "Variable",
                                "name": "site",
                                "variableName": "site"
                              }
                            ],
                            "kind": "ScalarField",
                            "name": "url",
                            "storageKey": null
                          },
                          {
                            "alias": null,
                            "args": null,
                            "kind": "ScalarField",
                            "name": "__typename",
                            "storageKey": null
                          }
                        ],
                        "storageKey": null
                      },
                      {
                        "alias": null,
                        "args": null,
                        "kind": "ScalarField",
                        "name": "cursor",
                        "storageKey": null
                      }
                    ],
                    "storageKey": null
                  },
                  {
                    "alias": null,
                    "args": null,
                    "concreteType": "PageInfo",
                    "kind": "LinkedField",
                    "name": "pageInfo",
                    "plural": false,
                    "selections": [
                      {
                        "alias": null,
                        "args": null,
                        "kind": "ScalarField",
                        "name": "endCursor",
                        "storageKey": null
                      },
                      {
                        "alias": null,
                        "args": null,
                        "kind": "ScalarField",
                        "name": "hasNextPage",
                        "storageKey": null
                      }
                    ],
                    "storageKey": null
                  }
                ],
                "storageKey": null
              },
              {
                "alias": null,
                "args": [
                  {
                    "kind": "Variable",
                    "name": "after",
                    "variableName": "cursor"
                  },
                  {
                    "kind": "Variable",
                    "name": "first",
                    "variableName": "count"
                  }
                ],
                "filters": null,
                "handle": "connection",
                "key": "PaginationFragment_friends",
                "kind": "LinkedHandle",
                "name": "friends"
              }
            ],
            "type": "Page",
            "abstractKey": null
          }
        ],
        "storageKey": null
      }
    ]
  },
  "params": {
    "cacheID": "31dcd57f42128b8d975e6224202a17c9",
    "id": null,
    "metadata": {},
    "name": "RefetchableFragmentQuery",
    "operationKind": "query",
    "text": null
  }
}

QUERY:

query RefetchableFragmentQuery(
  $count: Int = 10
  $cursor: ID
  $site: String
  $id: ID!
) {
  node(id: $id) {
    __typename
    ...prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment_1Xo9Zp
    id
  }
}

fragment prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment_1Xo9Zp on Page {
  name
  friends(after: $cursor, first: $count) {
    edges {
      ...prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment__edges_1Xo9Zp
    }
    pageInfo {
      endCursor
      hasNextPage
    }
  }
  id
}

fragment prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment__edges_1Xo9Zp on FriendsEdge {
  node {
    id
    url(site: $site)
    __typename
  }
  cursor
}


{
  "fragment": {
    "argumentDefinitions": [
      {
        "defaultValue": null,
        "kind": "LocalArgument",
        "name": "site"
      }
    ],
    "kind": "Fragment",
    "metadata": null,
    "name": "prefetchablePaginationQueryWithoutConflictingArgsQuery",
    "selections": [
      {
        "alias": null,
        "args": [
          {
            "kind": "Literal",
            "name": "id",
            "value": "x"
          }
        ],
        "concreteType": null,
        "kind": "LinkedField",
        "name": "node",
        "plural": false,
        "selections": [
          {
            "args": [
              {
                "kind": "Variable",
                "name": "site",
                "variableName": "site"
              }
            ],
            "kind": "FragmentSpread",
            "name": "prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment"
          },
          {
            "args": [
              {
                "kind": "Variable",
                "name": "site",
                "variableName": "site"
              }
            ],
            "kind": "FragmentSpread",
            "name": "prefetchablePaginationQueryWithoutConflictingArgs_SecondFragment"
          }
        ],
        "storageKey": "node(id:\"x\")"
      }
    ],
    "type": "Query",
    "abstractKey": null
  },
  "kind": "Request",
  "operation": {
    "argumentDefinitions": [
      {
        "defaultValue": null,
        "kind": "LocalArgument",
        "name": "site"
      }
    ],
    "kind": "Operation",
    "name": "prefetchablePaginationQueryWithoutConflictingArgsQuery",
    "selections": [
      {
        "alias": null,
        "args": [
          {
            "kind": "Literal",
            "name": "id",
            "value": "x"
          }
        ],
        "concreteType": null,
        "kind": "LinkedField",
        "name": "node",
        "plural": false,
        "selections": [
          {
            "alias": null,
            "args": null,
            "kind": "ScalarField",
            "name": "__typename",
            "storageKey": null
          },
          {
            "alias": null,
            "args": null,
            "kind": "ScalarField",
            "name": "id",
            "storageKey": null
          },
          {
            "kind": "InlineFragment",
            "selections": [
              {
                "alias": null,
                "args": null,
                "kind": "ScalarField",
                "name": "name",
                "storageKey": null
              },
              {
                "alias": null,
                "args": [
                  {
                    "kind": "Literal",
                    "name": "first",
                    "value": 10
                  }
                ],
                "concreteType": "FriendsConnection",
                "kind": "LinkedField",
                "name": "friends",
                "plural": false,
                "selections": [
                  {
                    "alias": null,
                    "args": null,
                    "concreteType": "FriendsEdge",
                    "kind": "LinkedField",
                    "name": "edges",
                    "plural": true,
                    "selections": [
                      {
                        "alias": null,
                        "args": null,
                        "concreteType": "User",
                        "kind": "LinkedField",
                        "name": "node",
                        "plural": false,
                        "selections": [
                          {
                            "alias": null,
                            "args": null,
                            "kind": "ScalarField",
                            "name": "id",
                            "storageKey": null
                          },
                          {
                            "alias": null,
                            "args": [
                              {
                                "kind": "Variable",
                                "name": "site",
                                "variableName": "site"
                              }
                            ],
                            "kind": "ScalarField",
                            "name": "url",
                            "storageKey": null
                          },
                          {
                            "alias": null,
                            "args": null,
                            "kind": "ScalarField",
                            "name": "__typename",
                            "storageKey": null
                          }
                        ],
                        "storageKey": null
                      },
                      {
                        "alias": null,
                        "args": null,
                        "kind": "ScalarField",
                        "name": "cursor",
                        "storageKey": null
                      }
                    ],
                    "storageKey": null
                  },
                  {
                    "alias": null,
                    "args": null,
                    "concreteType": "PageInfo",
                    "kind": "LinkedField",
                    "name": "pageInfo",
                    "plural": false,
                    "selections": [
                      {
                        "alias": null,
                        "args": null,
                        "kind": "ScalarField",
                        "name": "endCursor",
                        "storageKey": null
                      },
                      {
                        "alias": null,
                        "args": null,
                        "kind": "ScalarField",
                        "name": "hasNextPage",
                        "storageKey": null
                      }
                    ],
                    "storageKey": null
                  }
                ],
                "storageKey": "friends(first:10)"
              },
              {
                "alias": null,
                "args": [
                  {
                    "kind": "Literal",
                    "name": "first",
                    "value": 10
                  }
                ],
                "filters": null,
                "handle": "connection",
                "key": "PaginationFragment_friends",
                "kind": "LinkedHandle",
                "name": "friends"
              }
            ],
            "type": "Page",
            "abstractKey": null
          },
          {
            "kind": "InlineFragment",
            "selections": [
              {
                "alias": null,
                "args": [
                  {
                    "kind": "Literal",
                    "name": "first",
                    "value": 10
                  }
                ],
                "concreteType": "FriendsConnection",
                "kind": "LinkedField",
                "name": "friends",
                "plural": false,
                "selections": [
                  {
                    "alias": null,
                    "args": null,
                    "concreteType": "FriendsEdge",
                    "kind": "LinkedField",
                    "name": "edges",
                    "plural": true,
                    "selections": [
                      {
                        "alias": null,
                        "args": null,
                        "concreteType": "User",
                        "kind": "LinkedField",
                        "name": "node",
                        "plural": false,
                        "selections": [
                          {
                            "alias": null,
                            "args": null,
                            "kind": "ScalarField",
                            "name": "id",
                            "storageKey": null
                          },
                          {
                            "alias": null,
                            "args": [
                              {
                                "kind": "Variable",
                                "name": "site",
                                "variableName": "site"
                              }
                            ],
                            "kind": "ScalarField",
                            "name": "url",
                            "storageKey": null
                          },
                          {
                            "alias": null,
                            "args": null,
                            "kind": "ScalarField",
                            "name": "__typename",
                            "storageKey": null
                          }
                        ],
                        "storageKey": null
                      },
                      {
                        "alias": null,
                        "args": null,
                        "kind": "ScalarField",
                        "name": "cursor",
                        "storageKey": null
                      }
                    ],
                    "storageKey": null
                  },
                  {
                    "alias": null,
                    "args": null,
                    "concreteType": "PageInfo",
                    "kind": "LinkedField",
                    "name": "pageInfo",
                    "plural": false,
                    "selections": [
                      {
                        "alias": null,
                        "args": null,
                        "kind": "ScalarField",
                        "name": "endCursor",
                        "storageKey": null
                      },
                      {
                        "alias": null,
                        "args": null,
                        "kind": "ScalarField",
                        "name": "hasNextPage",
                        "storageKey": null
                      }
                    ],
                    "storageKey": null
                  }
                ],
                "storageKey": "friends(first:10)"
              },
              {
                "alias": null,
                "args": [
                  {
                    "kind": "Literal",
                    "name": "first",
                    "value": 10
                  }
                ],
                "filters": null,
                "handle": "connection",
                "key": "PaginationFragment2_friends",
                "kind": "LinkedHandle",
                "name": "friends"
              }
            ],
            "type": "User",
            "abstractKey": null
          }
        ],
        "storageKey": "node(id:\"x\")"
      }
    ]
  },
  "params": {
    "cacheID": "61508c3bf1b78d9d4b7d6609870e28da",
    "id": null,
    "metadata": {},
    "name": "prefetchablePaginationQueryWithoutConflictingArgsQuery",
    "operationKind": "query",
    "text": null
  }
}

QUERY:

query prefetchablePaginationQueryWithoutConflictingArgsQuery(
  $site: String
) {
  node(id: "x") {
    __typename
    ...prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment_QaI6y
    ...prefetchablePaginationQueryWithoutConflictingArgs_SecondFragment_QaI6y
    id
  }
}

fragment prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment_QaI6y on Page {
  name
  friends(first: 10) {
    edges {
      ...prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment__edges_2VKHkd
    }
    pageInfo {
      endCursor
      hasNextPage
    }
  }
  id
}

fragment prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment__edges_2VKHkd on FriendsEdge {
  node {
    id
    url(site: $site)
    __typename
  }
  cursor
}

fragment prefetchablePaginationQueryWithoutConflictingArgs_SecondFragment_QaI6y on User {
  id
  friends(first: 10) {
    edges {
      node {
        id
        url(site: $site)
        __typename
      }
      cursor
    }
    pageInfo {
      endCursor
      hasNextPage
    }
  }
}


{
  "argumentDefinitions": [
    {
      "defaultValue": 10,
      "kind": "LocalArgument",
      "name": "count"
    },
    {
      "defaultValue": null,
      "kind": "LocalArgument",
      "name": "cursor"
    },
    {
      "defaultValue": null,
      "kind": "LocalArgument",
      "name": "site"
    }
  ],
  "kind": "Fragment",
  "metadata": {
    "connection": [
      {
        "count": "count",
        "cursor": "cursor",
        "direction": "forward",
        "path": [
          "friends"
        ]
      }
    ],
    "refetch": {
      "connection": {
        "forward": {
          "count": "count",
          "cursor": "cursor"
        },
        "backward": null,
        "path": [
          "friends"
        ]
      },
      "fragmentPathInResult": [
        "node"
      ],
      "operation": require('RefetchableFragmentQuery.graphql'),
      "identifierInfo": {
        "identifierField": "id",
        "identifierQueryVariableName": "id"
      },
      "edgesFragment": require('prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment__edges.graphql')
    }
  },
  "name": "prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment",
  "selections": [
    {
      "alias": null,
      "args": null,
      "kind": "ScalarField",
      "name": "name",
      "storageKey": null
    },
    {
      "alias": "friends",
      "args": null,
      "concreteType": "FriendsConnection",
      "kind": "LinkedField",
      "name": "__PaginationFragment_friends_connection",
      "plural": false,
      "selections": [
        {
          "alias": null,
          "args": null,
          "concreteType": "FriendsEdge",
          "kind": "LinkedField",
          "name": "edges",
          "plural": true,
          "selections": [
            {
              "args": [
                {
                  "kind": "Variable",
                  "name": "count",
                  "variableName": "count"
                },
                {
                  "kind": "Variable",
                  "name": "cursor",
                  "variableName": "cursor"
                },
                {
                  "kind": "Variable",
                  "name": "site",
                  "variableName": "site"
                }
              ],
              "kind": "FragmentSpread",
              "name": "prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment__edges"
            }
          ],
          "storageKey": null
        },
        {
          "alias": null,
          "args": null,
          "concreteType": "PageInfo",
          "kind": "LinkedField",
          "name": "pageInfo",
          "plural": false,
          "selections": [
            {
              "alias": null,
              "args": null,
              "kind": "ScalarField",
              "name": "endCursor",
              "storageKey": null
            },
            {
              "alias": null,
              "args": null,
              "kind": "ScalarField",
              "name": "hasNextPage",
              "storageKey": null
            }
          ],
          "storageKey": null
        }
      ],
      "storageKey": null
    },
    {
      "alias": null,
      "args": null,
      "kind": "ScalarField",
      "name": "id",
      "storageKey": null
    }
  ],
  "type": "Page",
  "abstractKey": null
}

{
  "argumentDefinitions": [
    {
      "defaultValue": 10,
      "kind": "LocalArgument",
      "name": "count"
    },
    {
      "defaultValue": null,
      "kind": "LocalArgument",
      "name": "cursor"
    },
    {
      "defaultValue": null,
      "kind": "LocalArgument",
      "name": "site"
    }
  ],
  "kind": "Fragment",
  "metadata": {
    "plural": true
  },
  "name": "prefetchablePaginationQueryWithoutConflictingArgs_FirstFragment__edges",
  "selections": [
    {
      "alias": null,
      "args": null,
      "concreteType": "User",
      "kind": "LinkedField",
      "name": "node",
      "plural": false,
      "selections": [
        {
          "alias": null,
          "args": null,
          "kind": "ScalarField",
          "name": "id",
          "storageKey": null
        },
        {
          "alias": null,
          "args": [
            {
              "kind": "Variable",
              "name": "site",
              "variableName": "site"
            }
          ],
          "kind": "ScalarField",
          "name": "url",
          "storageKey": null
        },
        {
          "alias": null,
          "args": null,
          "kind": "ScalarField",
          "name": "__typename",
          "storageKey": null
        }
      ],
      "storageKey": null
    },
    {
      "alias": null,
      "args": null,
      "kind": "ScalarField",
      "name": "cursor",
      "storageKey": null
    }
  ],
  "type": "FriendsEdge",
  "abstractKey": null
}

{
  "argumentDefinitions": [
    {
      "defaultValue": 10,
      "kind": "LocalArgument",
      "name": "count"
    },
    {
      "defaultValue": null,
      "kind": "LocalArgument",
      "name": "cursor"
    },
    {
      "defaultValue": null,
      "kind": "LocalArgument",
      "name": "site"
    }
  ],
  "kind": "Fragment",
  "metadata": {
    "connection": [
      {
        "count": "count",
        "cursor": "cursor",
        "direction": "forward",
        "path": [
          "friends"
        ]
      }
    ]
  },
  "name": "prefetchablePaginationQueryWithoutConflictingArgs_SecondFragment",
  "selections": [
    {
      "alias": null,
      "args": null,
      "kind": "ScalarField",
      "name": "id",
      "storageKey": null
    },
    {
      "alias": "friends",
      "args": null,
      "concreteType": "FriendsConnection",
      "kind": "LinkedField",
      "name": "__PaginationFragment2_friends_connection",
      "plural": false,
      "selections": [
        {
          "alias": null,
          "args": null,
          "concreteType": "FriendsEdge",
          "kind": "LinkedField",
          "name": "edges",
          "plural": true,
          "selections": [
            {
              "alias": null,
              "args": null,
              "concreteType": "User",
              "kind": "LinkedField",
              "name": "node",
              "plural": false,
              "selections": [
                {
                  "alias": null,
                  "args": null,
                  "kind": "ScalarField",
                  "name": "id",
                  "storageKey": null
                },
                {
                  "alias": null,
                  "args": [
                    {
                      "kind": "Variable",
                      "name": "site",
                      "variableName": "site"
                    }
                  ],
                  "kind": "ScalarField",
                  "name": "url",
                  "storageKey": null
                },
                {
                  "alias": null,
                  "args": null,
                  "kind": "ScalarField",
                  "name": "__typename",
                  "storageKey": null
                }
              ],
              "storageKey": null
            },
            {
              "alias": null,
              "args": null,
              "kind": "ScalarField",
              "name": "cursor",
              "storageKey": null
            }
          ],
          "storageKey": null
        },
        {
          "alias": null,
          "args": null,
          "concreteType": "PageInfo",
          "kind": "LinkedField",
          "name": "pageInfo",
          "plural": false,
          "selections": [
            {
              "alias": null,
              "args": null,
              "kind": "ScalarField",
              "name": "endCursor",
              "storageKey": null
            },
            {
              "alias": null,
              "args": null,
              "kind": "ScalarField",
              "name": "hasNextPage",
              "storageKey": null
            }
          ],
          "storageKey": null
        }
      ],
      "storageKey": null
    }
  ],
  "type": "User",
  "abstractKey": null
}
